Cargo.toml

```diff
@@ -8,7 +8,12 @@ edition = "2018"
 repository = "https://github.com/rust-vmm/vm-fdt"
 keywords = ["fdt"]
 
+[dependencies]
+hashbrown = "0.14"
```

Comment:
We are trying to keep the dependencies to a minimum, and for this crate we have none. Is there a way to have this at least as an optional dependency?

Reply:
There is no _nice_ way of doing that since Cargo features are additive and `feature = "std"` would only add things.

We could add an `alloc` feature that has to be enabled explicitly. That way `cargo check --no-default-features` without any features will fail though, and one has to enable `alloc` explicitly on `no_std`.

Reply:
I guess the other alternative would be to just skip duplicate phandle checking in the `no_std` version so we would not need a `HashSet` at all; that would make it the caller's responsibility to ensure phandles are unique (like it was before we added this check). Maybe tying that to `no_std` would be too confusing, though.

Reply:
I think, I'd rather not have this compile on `no_std` without a feature than removing functionality from this crate. I'll implement this shortly.

Reply:
Are you okay with the solution I implemented? :)

Reply:
I think it's fine as it is, I was just wondering if it's possible.